librz/core/yank.c

```diff
@@ -94,30 +68,29 @@ static int perform_mapped_file_yank(RzCore *core, ut64 offset, ut64 len, const c
 	return res;
 }
 
-RZ_API int rz_core_yank_set(RzCore *core, ut64 addr, const ut8 *buf, ut32 len) {
-	// free (core->yank_buf);
-	if (buf && len) {
-		// FIXME: direct access to base should be avoided (use _sparse
-		// when you need buffer that starts at given addr)
-		rz_buf_set_bytes(core->yank_buf, buf, len);
-		core->yank_addr = addr;
-		return true;
-	}
-	return false;
+RZ_API bool rz_core_yank_set(RzCore *core, ut64 addr, const ut8 *buf, ut32 len) {
+	if (!buf || !len) {
+		return false;
+	}
+	// FIXME: direct access to base should be avoided (use _sparse
+	// when you need buffer that starts at given addr)
+	rz_buf_set_bytes(core->yank_buf, buf, len);
+	core->yank_addr = addr;
+	return true;
 }
 
 // Call set and then null terminate the bytes.
-RZ_API int rz_core_yank_set_str(RzCore *core, ut64 addr, const char *str, ut32 len) {
+RZ_API bool rz_core_yank_set_str(RzCore *core, ut64 addr, const char *str, ut32 len) {
 	// free (core->yank_buf);
-	int res = rz_core_yank_set(core, addr, (ut8 *)str, len);
+	bool res = rz_core_yank_set(core, addr, (ut8 *)str, len);
 	if (res == true) {
 		ut8 zero = 0;
 		rz_buf_write_at(core->yank_buf, len - 1, &zero, sizeof(zero));
 	}
 	return res;
 }
 
-RZ_API int rz_core_yank(struct rz_core_t *core, ut64 addr, int len) {
+RZ_API bool rz_core_yank(struct rz_core_t *core, ut64 addr, int len) {
```

Comment:
i'm wondering why this function needs to use `rz_core_seek`... Do you know? I'm trying to add a bunch of simple unit tests for yank API.